5-network/03-fetch-progress/article.md

```diff
@@ -1,60 +1,60 @@
+W przeciwieństwie do `response.text()`, `response.json()` czy innych metod, `response.body` pozwala na całkowitą kontrolę nad procesem odczytu, co pozwala na określenie, jaka ilość danych jest zużywana w dowolnym momencie.
 
-Here's the sketch of code that reads the reponse from `response.body`:
+Oto przykład kodu, który odczytuje odpowiedź z `response.body`:
 
 ```js
-// instead of response.json() and other methods
+// zamiast response.json() i innych metod
 const reader = response.body.getReader();
 
-// infinite loop while the body is downloading
+// pętla nieskończona w momencie pobierania ciała odpowiedzi
 while(true) {
-  // done is true for the last chunk
-  // value is Uint8Array of the chunk bytes
+  // done przyjmuje wartość true dla ostatniego kawałka
+  // value jest tablicą Uint8Array bajtów danego kawałka  
   const {done, value} = await reader.read();
 
   if (done) {
     break;
   }
 
-  console.log(`Received ${value.length} bytes`)
+  console.log(`Pobrano ${value.length} bajtów`)
 }
 ```
 
-The result of `await reader.read()` call is an object with two properties:
-- **`done`** -- `true` when the reading is complete, otherwise `false`.
-- **`value`** -- a typed array of bytes: `Uint8Array`.
+Rezultatem wywołania `await reader.read()` jest obiekt, posiadający dwie właściwości:
+- **`done`** -- `true` po zakończeniu odczytu, w przeciwnym przypadku `false`.
+- **`value`** -- reprezentująca typ tablica bajtów `Uint8Array`.
 
 ```smart
-Streams API also describes asynchronous iteration over `ReadableStream` with `for await..of` loop, but it's not yet widely supported (see [browser issues](https://github.com/whatwg/streams/issues/778#issuecomment-461341033)), so we use `while` loop.
+Streams API opisuje też asynchroniczną iterację po `ReadableStream` za pomocą pętli `for await..of`, aczkolwiek to rozwiązanie nie jest szeroko wspierane (zob. [problemy z przeglądarką](https://github.com/whatwg/streams/issues/778#issuecomment-461341033)), wobec tego używamy pętli `while`.
 ```
 
-We receive response chunks in the loop, until the loading finishes, that is: until `done` becomes `true`.
+Otrzymujemy kawałki odpowiedzi w pętli, aż do zakończenia ładowania, to znaczy dopóki `done` nie stanie się `true`.
 
-To log the progress, we just need for every received fragment `value` to add its length to the counter.
+Aby rejestrować postęp, wystarczy dodawać do licznika długość tablicy `value` każdego otrzymanego kawałka.
 
-Here's the full working example that gets the response and logs the progress in console, more explanations to follow:
+Oto w pełni działający przykład, w którym postęp otrzymywanej odpowiedzi jest wyświetlany w konsoli. Szczegóły w dalszej części artykułu.
 
 ```js run async
-// Step 1: start the fetch and obtain a reader
+// Krok 1: uruchom pobieranie i podepnij czytnik
```

Comment:
```suggestion
// Krok 1: Uruchom pobieranie i podepnij czytnik
```

Reply:
Miałem tak w pierwotnej wersji, ale jakoś do mnie nie przemawiało. Ale skoro i tobie tak wyszło, to może coś w tym jest.